pkg/proctree/proctree.go

```diff
@@ -0,0 +1,118 @@
+package proctree
+
+import (
+	"os"
+
+	"github.com/aquasecurity/tracee/pkg/errfmt"
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/pkg/proctree/nodes"
+	"github.com/aquasecurity/tracee/pkg/proctree/tree"
+	"github.com/aquasecurity/tracee/pkg/utils"
+)
+
+type ProcTree struct {
+	tree *tree.Tree
+}
+
+func NewProcTree() *ProcTree {
+	// TODO: read from ProcFS from time to time
```

Comment:
Mitigate possible event losses and keep tree more accurate. Also, mitigate the initial load race we've discussed. Giving a second thought, if we update the process tress from the regular pipeline, than even if we miss events in the control plane (or there) we will always have a chance to update the tree with other events. So, maybe it wont be needed.

Reply:
I think that it will be good either way, because in the end we can't really know what load we will face in some environments which might result many misinformation in the tree.
Having it double checked using the procfs sounds legit, and not performance-heavy to do.